src/components/bookings/SeatingSeat.vue

```diff
@@ -1,27 +1,42 @@
 <script setup lang="ts">
 import { computed } from "vue";
 import { useBookingStore } from "@/stores/booking";
+
 const bookingStore = useBookingStore();
-defineProps<{
+
+const props = defineProps<{
   letter: string;
   seat: string;
 }>();
-const emit = defineEmits(["addSeat"]);
+const emit = defineEmits<{
+  (e: "addSeat", id: string): void;
+}>();
 
 const storeSelectedTickets = computed(() => {
   return bookingStore.selectedTickets.map((ticket) => ticket.seat);
 });
+const seatValue = computed(() => {
+  return (letter: string, seat: string) => letter + seat;
+});
+const isSeatTaken = computed(() => {
+  return (letter: string, seat: string) =>
+    bookingStore.selectedReservation.taken_seats.includes(
+      seatValue.value(letter, seat)
+    );
+});
+const isSeatSelected = computed(() => {
+  return (letter: string, seat: string) =>
+    storeSelectedTickets.value.includes(seatValue.value(letter, seat));
+});
```

Comment:
That is not really how you should use computed. It should not return a function for you to use. in seatValue you can just `return props.letter + props.seat` and in following computed you just reference seatValue.value